src/shogun/distributions/classical/GaussianDistributionNew.h

```diff
@@ -0,0 +1,175 @@
+
+	CGaussianDistributionNew(SGVector<float64_t> mean, SGVector<float64_t> cov,
+			bool cov_is_cholesky=false);
+
+	CGaussianDistributionNew(SGVector<float64_t> mean, SGMatrix<float64_t> cov,
+			bool cov_is_cholesky=false);
+
+	/** Destructor */
+	virtual ~CGaussianDistributionNew();
+
+	void set_mean(SGVector<float64_t> cov);
+	void set_cov(float64_t cov);
+	void set_cov(SGVector<float64_t> cov);
+	void set_cov(SGMatrix<float64_t> cov);
+
+	SGVector<float64_t> get_mean() const;
+	float64_t get_cov_spherical() const;
+	SGVector<float64_t> get_cov_diag() const;
+	SGMatrix<float64_t> get_cov_full() const;
+
+	/** Samples from the distribution multiple times
+	 *
+	 * @param num_samples number of samples to generate
+	 * @param pre_samples a matrix of standard normal samples that might be used
+	 * for sampling the Gaussian. Ignored by default. If passed, the pre-samples
+	 * will be modified.
+	 * @return matrix with samples (column vectors)
+	 */
+	virtual SGMatrix<float64_t> sample(int32_t num_samples,
+			SGMatrix<float64_t> pre_samples=SGMatrix<float64_t>()) const;
+
+	/** Computes the log-pdf for all provided samples. That is
+	 *
+	 * \f[
+	 * \log(\mathcal{N}_x(\mu,\Sigma))=
+	 * - \frac{d}{2}  \log(2\pi)
+	 * -\frac{1}{2}\log(\det(\Sigma))
+	 * -\frac{1}{2}(x-\mu)^T\Sigma^{-1}(x-\mu),
+	 * \f]
+	 *
+	 * where \f$d\f$ is the dimension of the Gaussian.
+	 * The method to compute the log-determinant is based on the factorization
+	 * of the covariance matrix.
+	 *
+	 * @param samples samples to compute log-pdf of (column vectors)
+	 * @return vector with log-pdfs of given samples
+	 */
+
+	virtual SGVector<float64_t> log_pdf_multiple(SGMatrix<float64_t> samples) const;
+
+	/** @return name of the SGSerializable */
+	virtual const char* get_name() const
+	{
+		return "GaussianDistribution";
+	}
+
+
+	/** Computes the univariate pdf for one given sample.
+	 *
+	 * @param sample is a given sample
+	 * @param mu is the mean of univariate Normal distribution (default value is 0.0)
+	 * @param sigma2 is the variance of univariate Normal distribution (default value is 1.0)
+	 * @return the pdf of the distribution given the sample
+	 */
+	static float64_t univariate_log_pdf(float64_t sample, float64_t mu = 0.0, float64_t sigma2 = 1.0)
+	{
+		REQUIRE(sigma2 > 0, "Variance should be positive\n");
+		return -0.5 * (CMath::pow(sample - mu, 2) / sigma2
+			+ CMath::log(2.0 * CMath::PI) + CMath::log(sigma2));
+	}
+
+//	static float64_t multivariate_log_pdf(SGVector<float64_t> sample, SGVector<float64_t> mu, SGVector<float64_t> cov)
+//	{
+//		// TODO:
+//		REQUIRE(sigma2 > 0, "Variance should be positive\n");
+//		return -0.5 * (CMath::pow(sample - mu, 2) / sigma2
+//			+ CMath::log(2.0 * CMath::PI) + CMath::log(sigma2));
+//	}
+
+private:
+
+	/** Initialses and registers parameters */
+	void init();
+
+protected:
+	/** Mean */
+	SGVector<float64_t> m_mean;
+
+	/** Lower factor of covariance matrix (depends on factorization type).
```

Comment:
This should be updated and explain the different parametrisations possible here, and what the matrix contains in the case and which dimensions it has